qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500.py

```diff
@@ -192,5 +192,24 @@ def use_nplc_for_high_resolution_adc(
             coeff=n
         )
 
+    def use_manual_mode_for_high_speed_adc(
+            self, n: Optional[Union[int, float]] = None) -> None:
+        """
+        Set the high-speed ADC to manual mode, with optionally defining number
+        of averaging samples via argument `n`.
+
+        Use ``n=1`` to disable averaging (``n=None`` uses the default
+        setting from the instrument which is also ``n=1``).
+
+        Args:
+            n: Number of averaging samples, between 1 and 1023. Default
+                setting is 1. (For more info see Table 4-21.)
+        """
+        self._setup_integration_time(
+            adc_type=constants.AIT.Type.HIGH_SPEED,
+            mode=constants.AIT.Mode.MANUAL,
+            coeff=n
+        )
```

Comment:
Is there a supported (non private) way to set this back to automatic?

Reply:
At the moment no. However, if this issue crops up (some user requests for it) we can make _setup_integration_time public.